protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -269,7 +269,7 @@ func nestedQueryParams(message *descriptor.Message, field *descriptor.Field, pre
 		// verify if the field is required in message options
 		if messageSchema, err := extractSchemaOptionFromMessageDescriptor(message.DescriptorProto); err == nil {
 			for _, fieldName := range messageSchema.GetJsonSchema().GetRequired() {
-				if fieldName == reg.FieldName(field) {
+				if fieldName == reg.FieldName(field) || fieldName == field.GetName() {
```

Comment:
What is the case where `reg.FieldName(field)` and `field.GetName()` would differ?

Reply:
`reg.FieldName()` returns `field.GetName()` or `field.GetJsonName()`, depending on the flag `json_names_for_fields`. The condition could also be replaced with `fieldName == field.GetName() || fieldName == field.GetJsonName()`

Reply:
Thanks for the clarification! I think the latter suggestion is easier to understand when reading the code.

As I understand this, then for e.g. field `foo_bar`, `required` could be set to either `foo_bar` or `fooBar` (or a custom JSON name) with the same result. I don't see any problem with providing this flexibility for `required`.

Reply:
Replaced with a check against field.GetName() and field.GetJsonName()

Reply:
Looks good to me!

Reply:
@patrick246 And thanks for fixing this issue.